src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs

```diff
@@ -82,13 +86,43 @@ internal InfiniteScrollList(Lazy<JoinableTaskFactory> joinableTaskFactory)
 
             BindingOperations.EnableCollectionSynchronization(Items, _list.ItemsLock);
 
-            ItemsView = new CollectionViewSource() { Source = Items }.View;
-            DataContext = ItemsView;
+            ItemsView = new CollectionViewSource() { Source = Items };
+            ListCollectionView itemsView = ItemsView.View as ListCollectionView;
+            itemsView.IsLiveFiltering = true;
+            itemsView.IsLiveGrouping = true;
+            itemsView.LiveFilteringProperties.Add($"{nameof(PackageItemViewModel.IsPackageVulnerable)}");
+            itemsView.LiveGroupingProperties.Add($"{nameof(PackageItemViewModel.PackageLevel)}");
+            itemsView.Filter = item =>
+            {
+                if (item is not null && item.Equals(_loadingVulnerabilitiesStatusIndicator))
+                {
+                    if (FilterByVulnerabilities)
+                    {
+                        return true;
+                    }
+                    return false;
+                }
+
+                PackageItemViewModel vitem = item as PackageItemViewModel;
+                return IsItemInFilter(vitem);
```

Comment:
More concise, avoids variable declaration that is only used once.

```suggestion
                return IsItemInFilter(item as PackageItemViewModel);
```

Reply:
This was marked as resolved but not addressed. Please leave comment or fix. Thanks!